src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -661,17 +662,29 @@ public void addEnvVars(List<TemplateEnvVar> envVars) {
         }
     }
 
+    @SuppressWarnings("unused") // Used by jelly
+    @Restricted(DoNotUse.class) // Used by jelly
+    public boolean hasManagePermission() {
+        StaplerRequest2 request = Stapler.getCurrentRequest2();
+        if (request != null) {
+            PodTemplateGroup groupFromRequest = request.findAncestorObject(PodTemplateGroup.class);
+            if(groupFromRequest!=null) {
+                return groupFromRequest.hasManagePermission();
+            };
+        }
+        return Jenkins.get().hasPermission(Jenkins.MANAGE);
+    }
+
     /**
      * Deletes the template.
      */
     @POST
     public HttpResponse doDoDelete(@AncestorInPath PodTemplateGroup owner) throws IOException {
-        Jenkins j = Jenkins.get();
-        j.checkPermission(Jenkins.MANAGE);
         if (owner == null) {
             throw new IllegalStateException("Cloud could not be found");
         }
         owner.removeTemplate(this);
```

Comment:
Permission check being done in the `PodTemplateGroup`. I think it is risky to leave it to the implementer, it should better be directly implemented in `PodTemplateGroup`

Reply:
I am unsure what you are asking and I think I may misunderstood your first comment on this. `PodTemplateGroup`  is an interface, so how would you do this?